python_modules/jupedsim/jupedsim/distributions.py

```diff
@@ -1,2 +1,282 @@
+    if len(circle_segment_radii) != len(fill_parameters):
+        raise IncorrectParameterError(f"the number of circle segments does not match the number of fill parameters.\n"
+                                      f"radii given for {len(circle_segment_radii)} circle segments,"
+                                      f"fill parameter given for {len(fill_parameters)} circle segments")
+    for i, c_s_radius in enumerate(circle_segment_radii):
+        if c_s_radius[0] < 0 or c_s_radius[1] < 0:
+            raise NegativeValueError(f"Circle segment {c_s_radius[0]} : {c_s_radius[1]} is expected to be positiv")
+        if c_s_radius[0] >= c_s_radius[1]:
+            raise OverlappingCirclesError(f"inner radius bigger than/equal to outer radius\n"
+                                          f"a Circle segment from {c_s_radius[0]} to {c_s_radius[1]} is not possible")
+        j = 0
+        while j < i:
+            if c_s_radius[0] < c_s_radius[1] <= circle_segment_radii[j][0] \
+                    or circle_segment_radii[j][1] <= c_s_radius[0] < c_s_radius[1]:
+                j = j + 1
+                continue
+            else:
+                raise OverlappingCirclesError(f"the new Circle would overlap with"
+                                              f"the existing circle from {c_s_radius[0]} to {c_s_radius[1]}")
+
+
+def distribute_in_circles_by_number(*, polygon, distance_to_agents, distance_to_polygon,
+                                    center_point, circle_segment_radii, numbers_of_agents,
+                                    seed=None, max_iterations=10_000):
+    """returns points randomly placed inside the polygon inside each the circle segments
+
+        :param polygon: shapely polygon in which the agents will be placed
+        :param distance_to_agents: minimal distance between the centers of agents
+        :param distance_to_polygon: minimal distance between the center of agents and the polygon edges
+        :param center_point: the Center point of the circle segments
+        :param circle_segment_radii: a list of minimal and maximal radius for each circle segment
+            Circle segments must not overlap
+            formatted like [(minimum_radius, maximum_radius)]
+        :param numbers_of_agents: a list of number of agents for each Circle segment
+            the position of the number corresponds to the order in which the Circle segments are given
+        :param seed: define a seed for random generation, Default value is None which corresponds to a random value
+        :param max_iterations: no more than max_iterations must find a point inside the polygon, Default is 10_000
+        :return: list of created points"""
+
+    # catch wrong inputs
+    __catch_wrong_inputs(polygon=polygon, center_point=center_point,
+                         circle_segment_radii=circle_segment_radii, fill_parameters=numbers_of_agents)
+    if seed is not None:
+        np.random.seed(seed)
```

Comment:
if the seed is None there was no furter specified seed and the seed will be random.
With this changes it would be nessesary to select a specified seed. 
Should i change this?

Reply:
You have defined the default value `None` in the function signature.
`np.random.seed` uses  a random seed as default as well.
So the `if` clause is not actually required, right?

Reply:
that is right.
done